### PR TITLE
Adding goclient for discovery job

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: clients/discovery.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	respData, _ := c.GetBodyData(response.Body)
+	omeDiscoveryJob := models.DiscoveryJob{}
+	err = c.JSONUnMarshal(respData, &omeDiscoveryJob)
+	return omeDiscoveryJob, err
+}
+
+func (c *Client) UpdateDiscoveryJob(discoveryJob models.DiscoveryJobPayload) (models.DiscoveryJob, error) {
+	data, _ := c.JSONMarshal(discoveryJob)
+	x := DiscoveryJobAPI + "?groupId=" + fmt.Sprint(discoveryJob.DiscoveryConfigGroupID)
+	response, err := c.Post(x, nil, data)
```
Comment (rounak-adhikary) [inline clients/discovery.go:23]: Please try to use the Post method's query params attribute 
```suggestion
	response, err := c.Post(DiscoveryJobAPI, map[string]string{
	        "groupId": discoveryJob.DiscoveryConfigGroupID,
	}, data)
```

File: clients/discovery_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+func TestClient_DiscoveryCreateJob(t *testing.T) {
+	ts := createNewTLSServer(t)
+	defer ts.Close()
+
+	opts := initOptions(ts)
+
+	c, _ := NewClient(opts)
+	var createDiscoveryJobPayloadSuccess models.DiscoveryJobPayload
+	jsonData, err := ioutil.ReadFile("json_data/payloadCreateDiscovery.json")
```
Comment (rounak-adhikary) [inline clients/discovery_test.go:19]: Can we use embed for the json data?

File: clients/discovery.go (hunk 1 above)
Comment (Krishnan-Priyanshu) [reply clients/discovery.go]: add to group id to post method query params

File: clients/discovery_test.go (hunk 2 above)
Comment (Krishnan-Priyanshu) [reply clients/discovery_test.go]: add the embed package for json.

File: clients/discovery_test.go [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+	_ "embed"
 	"terraform-provider-ome/models"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
+var (
+	//go:embed json_data/payloadCreateDiscovery.json
+	jsonData1 []byte
```
Comment (rounak-adhikary) [inline clients/discovery_test.go:13]: Please give these variables proper names (the name of the file being embedded is a good starting point). All else is good.

Comment (Krishnan-Priyanshu) [reply clients/discovery_test.go:13]: "payloadCreateDiscovery []byte"
does this look good?

Comment (rounak-adhikary) [reply clients/discovery_test.go:13]: sure sure

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Review by Krunal-Thakkar — APPROVED** [review]
Please remove coverage.out from next PR.

**Review by prajwalpatil25 — APPROVED** [review]

**Pull request merged**